### PR TITLE
New question: Does this specification allow an origin to send data to the underlying platform? Fixes #96.

### LINKED ISSUE #96: Lessons from sharing URLs
I think we learned some good lessons from:

https://blog.redteam.pl/2020/08/stealing-local-files-using-safari-web.html

In particular, we can't assume that the OS or receiving application will safely handle random URL schemes. 

For APIs that pass URLs to other apps or the OS, the questionnaire should ask what will happen if the browser passes along "file://" or other URL schemes ("data://") - and what the potential risks are when the receiving application ingests and dereferences those URLs, which can lead to information leakage, data theft, bypassing firewalls, etc. 

As such, it would be good to add something to the questionnaire around URLs... URLs shouldn't be assumed to be safe strings. And that a URL itself doesn't represent a final destination, because once dereferenced a URL can redirect virtually anywhere.

### PR DESCRIPTION
**Pull request by hober**
Fixes #96.

***
<a href="https://pr-preview.s3.amazonaws.com/w3ctag/security-questionnaire/pull/99.html" title="Last updated on Jan 29, 2021, 12:15 AM UTC (55142f5)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3ctag/security-questionnaire/99/48c8ce7...55142f5.html" title="Last updated on Jan 29, 2021, 12:15 AM UTC (55142f5)">Diff</a>

### CODE REVIEW

**Comment by hober** [comment]
cc @marcoscaceres

File: index.bs [hunk 1]
```diff
@@ -397,6 +397,30 @@ changes for most users. Some user agents have taken steps to reduce the
...
 
+<h3 class=question id=send-to-platform>
+  Does this specification allow an origin to send data to the underlying
+  platform?
+</h3>
+
+If so, what kind of data can be sent?
+
+Platforms differ in how they process data passed into them,
+which may present different risks to users.
```
Comment (marcoscaceres) [inline index.bs:408]:
```suggestion
which may present different risks to users.

Don't assume the underlying platform will safely handle the data that is passed. 
Where possible, mitigate attacks by limiting or structuring the kind of data is passed to the platform.    
```

File: index.bs [hunk 2]
```diff
@@ -397,6 +397,30 @@ changes for most users. Some user agents have taken steps to reduce the
...
+and if they are dereferenced,
+redirects may or may not be followed.
+If your specification sends URLs to underlying platform APIs,
+the potential harm of *your* API
+may vary depending on
+the behavior of the various underlying platform APIs it's built upon.
+What happens when `file:`, `data:`, or `blob:` URLs
+are passed to the underlying platform API?
+Keep in mind that, even if your API only allows `http:` and `https:` URLs,
+servers can redirect these to  `file:`, `data:`, or `blob:` URLs.
```
Comment (marcoscaceres) [inline index.bs:421]:
```suggestion
servers can redirect these to  `file:`, `data:`, or `blob:` URLs,
which can potentially read data directly form the user's hard disk or from memory.
```

File: index.bs [hunk 3]
```diff
@@ -397,6 +397,30 @@ changes for most users. Some user agents have taken steps to reduce the
...
+URLs may or may not be dereferenced by a platform API,
+and if they are dereferenced,
+redirects may or may not be followed.
+If your specification sends URLs to underlying platform APIs,
+the potential harm of *your* API
+may vary depending on
+the behavior of the various underlying platform APIs it's built upon.
+What happens when `file:`, `data:`, or `blob:` URLs
+are passed to the underlying platform API?
+Keep in mind that, even if your API only allows `http:` and `https:` URLs,
```
Comment (martinthomson) [inline index.bs:420]: The implication here seems to be that "http[s]:" is safe.  That's not guaranteed either.  The webshare case showed that the position of a requester is what matters.  Sure, "file:" carries more inherent risk, but only in the aggregate; fetching from "https:" is no less problematic if the resource uses ambient authority to authorize access.

Comment (hober) [reply index.bs]: @martinthomson, is 68130bf any better?

Comment (martinthomson) [reply index.bs]: Much.  Thanks.

**Comment by hober** [comment]
I've adopted several changes based on @marcoscaceres & @martinthomson's feedback. @pes10k, I think this one is good to go now. What say you?

**Comment by pes10k** [comment]
I think this looks great!  Thanks @hober !

**Review by pes10k — APPROVED** [review]
looks great to me!

**Review by cynthia — APPROVED** [review]
LGTM2.

**Pull request merged**